node/lease_controller_v1.go

```diff
@@ -0,0 +1,341 @@
+package node
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
```

Comment:
Am I correct in assuming this is a verbatim copy? If so, couldn't we simply import this directly from the kubelet code instead of copying? Would make it easier for bumping changes.

Reply:
It's about 80% verbatim. There's a bunch of small changes, like using log vs. klog, or adding tracing.

Reply:
What happens when this changes upstream? How does one identify what changed and what not? Maybe commit the file as is upstream (one commit) and then commit changes so we are maybe able to rebase in the future?

Anyways, at least point out that there were changes and what they are?